subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/BeanNode.java

```diff
@@ -105,10 +105,11 @@ public BeanNode apply(@Nullable Object input) {
                 }
 
                 private String determinePropertyName(@Nullable Object input) {
+                    count++;
                     if (input instanceof Named) {
                         return ((Named) input).getName();
                     }
-                    return "$" + count++;
+                    return "$" + count;
```

Comment:
Let's not do it like this. Starting with `-1` is just wrong.

```java
private String determinePropertyName(@Nullable Object input) {
    String name;
    if (input instanceof Named) {
        name = ((Named) input).getName();
    } else {
        name = "$" + count;
    }
    count++;
    return name;
}
```

Reply:
Done.